libjsonutils/src/json_utils.cpp

```diff
@@ -1,23 +1,24 @@
+#include "file_utils.h"
+
 #include "jsonutils/json_utils.h"
 
 #include <boost/regex.hpp>
 #include <iostream>
 
-#include <file_utils.h>
 
 namespace jsonutils
 {
     //Extremely poor regex pattern for matching urls
     constexpr auto URL_PATTERN = "(http|https)://(\\w+\\.)+(\\w)/?(\\w+/{0,1})*";
 
-    boost::optional<rapidjson::Document> loadFromUrl(const std::string& url)
+    boost::optional<rapidjson::Document> loadFromUrl(const std::string& /* url */)
     {
         //Download resource and extract message body
         //Return boost::none if something went wrong, e.g. status code >= 300
         rapidjson::Document doc;
         doc.Parse("{\"source\": \"url\"}");
 
-        return doc;
+        return std::move(doc);
```

Comment:
These changes seem like they belong in a separate commit at least, probably a separate PR even.

Reply:
Yes, but without this, it does't compile (copy constructor is deleted in boost now).
I changed it to continue testing with cmake.

Reply:
which version of boost, rapidjson and compiler are you using? this compiles just fine in my system

Reply:
alright I was able to find the problem. This has nothing to do with `boost::optional` (which is both a copiable and movable type), but rather `rapidjson::Document`, which is a move only type. Modern compilers like mine (g++9.2.1) are smart enough to move the document into the `boost::optional`, yours doesn't and tries to copy it into the optional.

Reply:
```suggestion
       // force-move into the optional, as some compiler's don't do this automatically
        return std::move(doc);
```

Reply:
done
Claus-MBP:example_exec clausklein$ clang --version
clang version 9.0.0 (trunk 365158)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /opt/local/libexec/llvm-devel/bin